ethsigner/commandline/src/main/java/tech/pegasys/ethsigner/EthSignerBaseCommand.java

```diff
@@ -107,22 +108,43 @@ public File getStorePasswordFile() {
     }
   }
 
-  static class TlsClientAuthentication {
+  static class TlsAuthorisationMechanisms implements ClientAuthConstraints {
+
     @Option(
         names = "--tls-known-clients-file",
         description = "Path to a file containing the fingerprints of authorized clients.",
-        arity = "1",
-        required = true)
+        arity = "1")
     private File tlsKnownClientsFile = null;
 
+    @Option(
+        names = "--tls-allow-ca-clients",
+        description = "If defined, allows clients authorised by the CA to connect to Ethsigner.",
+        arity = "0")
+    private Boolean tlsAllowCaClients = false;
+
+    @Override
+    public Optional<File> getKnownClientsFile() {
+      return Optional.ofNullable(tlsKnownClientsFile);
+    }
+
+    @Override
+    public boolean isCaAuthorisedClientAllowed() {
+      return tlsAllowCaClients;
+    }
+  }
+
+  static class TlsClientAuthentication {
+
     @SuppressWarnings("UnusedVariable")
+    @ArgGroup(exclusive = false)
+    private TlsAuthorisationMechanisms authMechanisms;
+
     @Option(
         names = "--tls-allow-any-client",
         description =
-            "If defined, will allow any client to connect. Is mutually exclusive with "
-                + "--tls-known-clients-file.",
-        arity = "0",
-        required = false)
+            "If defined, will allow any client to connect. Is mutually exclusive with other "
+                + "client auth settings",
```

Comment:
nit: full spelling for auth to make it clear it's authentication?

Reply:
done